tests/test_test_tools/test_test_contract.py

```diff
@@ -72,14 +72,14 @@ def setup(self) -> None:
         """Setup test"""
 
         # must `copy` the class to avoid test interference
-        class_copy = type("TestCls", TestCls.__bases__, dict(TestCls.__dict__))
-        self.test_cls = cast(TestCls, class_copy)
+        self.test_cls = cast(TestCls, copy_class(TestCls))
 
     def setup_test_cls(self) -> TestCls:
         """Helper method to setup test to be tested"""
 
-        self.test_cls.setup()
-        return self.test_cls()  # type: ignore
+        test_instance = self.test_cls()  # type: ignore
+        test_instance.setup()
+        return test_instance
```

Comment:
changes this because this is how it should be done
1. `setup_class()`
2. instantiate
3. `setup()`